tests/integration/mixer/policy/ratelimiting_test.go

```diff
@@ -112,6 +117,57 @@ func TestRateLimiting_DefaultLessThanOverride(t *testing.T) {
 		})
 }
 
+func TestRateLimiting_ExternalTraffic(t *testing.T) {
+	framework.Run(t, func(ctx framework.TestContext) {
+		destinationService := "istio-egressgateway"
+
+		bookInfoNameSpaceStr := bookinfoNs.Name()
+		config := setupConfigOrFail(t, bookinfo.CNNRedisRateLimit, bookInfoNameSpaceStr,
+			red, g, ctx)
+		g.ApplyConfigOrFail(t,
+			bookinfoNs,
+			strings.Replace(egressConfig, "<bookinfo_namespace>", bookInfoNameSpaceStr, -1))
+		defer func() {
+			deleteConfigOrFail(t, config, g, ctx)
+			g.DeleteConfigOrFail(t,
+				bookinfoNs,
+				strings.Replace(egressConfig, "<bookinfo_namespace>", bookInfoNameSpaceStr, -1))
+			util.AllowRuleSync(t)
+		}()
+		util.AllowRuleSync(t)
+
+		res, err := RunCurlTest(300, "http://edition.cnn.com/politics", bookInfoNameSpaceStr, ctx)
+		if err != nil {
+			t.Fatalf("error curling %v", err)
+		}
+
+		want200s := 50.0
+		// Need 50 200s, allowing for some randomness in routing
+		want200s = math.Floor(want200s * 0.70)
+		ns := namespace.ClaimSystemNamespaceOrFail(t, ctx)
+		if res.got200 < want200s {
+			attributes := []string{fmt.Sprintf("%s=\"%s\"", util.GetDestinationLabel(),
+				util.Fqdn(destinationService, ns.Name())),
+				fmt.Sprintf("%s=\"%d\"", util.GetResponseCodeLabel(), 200),
+				fmt.Sprintf("%s=\"%s\"", util.GetReporterCodeLabel(), "source")}
+			t.Logf("prometheus values for istio_requests_total for 200's:\n%s",
+				util.PromDumpWithAttributes(prom, "istio_requests_total", attributes))
```

Comment:
Why do we need to check the request metrics, can we just see how many requests were successful directly from the client responses?

Reply:
This is in the case when we fail the test... For this test we are using response from client to see if it's successful or not

Reply:
I see, makes sense